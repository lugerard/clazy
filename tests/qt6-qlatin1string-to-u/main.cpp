#include <QtCore/QString>

int totototo = 112233;

void receivingQString(QString s1) {}
void receivingQLatin1String(QLatin1String s1) {}


#define PREFIX "foo"
void test()
{
    QString s1 = QLatin1String("str");
    QString s2 = QString(QLatin1String("s2"));
    s1 += QLatin1String("str");
    s1 = QLatin1String(true ? "foo" : "bar");
    s1.append(QLatin1String("appending"));

    receivingQString( QLatin1String("str"));
    receivingQLatin1String( QLatin1String("latin"));

    QLatin1String toto = QLatin1String("toto");

    const char* myChar = "foo";
    QString ww = QLatin1String(myChar); // should not be fixed
    QString string_with_macro = QLatin1String(PREFIX "bar"); // should not be fixed!!! (the test will fail)

}
