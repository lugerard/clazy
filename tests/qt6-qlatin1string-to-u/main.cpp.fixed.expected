#include <QtCore/QString>

int totototo = 112233;

void receivingQString(QString s1) {}
void receivingQLatin1String(QLatin1String s1) {}


#define PREFIX "foo"
void test()
{
    QString s1 = u"str";
    QString s2 = QString(u"s2");
    s1 += u"str";
    s1 = true ? u"foo" : u"bar";
    s1.append(u"appending");

    receivingQString( u"str");
    receivingQLatin1String( u"latin");

    QLatin1String toto = u"toto";

    const char* myChar = "foo";
    QString ww = QLatin1String(myChar); // should not be fixed
    QString string_with_macro = QLatin1String(PREFIX "bar"); // should not be fixed!!! (the test will fail)

}
